cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -867,8 +867,24 @@ class struct_scalar : public scalar {
  private:
   table _data;
 
-  void init(bool is_valid, rmm::cuda_stream_view stream, rmm::mr::device_memory_resource* mr);
-  void superimpose_nulls(rmm::cuda_stream_view stream, rmm::mr::device_memory_resource* mr);
+  /**
+   * @brief Check if all the input columns constructing this struct scalar have valid size.
+   */
+  void check_size();
```

Comment:
This is where we assert a class invariant. It would better named as `assert_valid_size()`.